### DISCUSSION TITLE
I want a method to compress the large size image to a fixed smaller size before saving it to cash directory.

### DISCUSSION THREAD

**Discussion by MahmoudAlEssawi** [dotnet/maui#16746] [Q&A]
I searched a lot for that but nothing worked

**Comment by xtuzy** [comment]
cash directory? Maui.Graphics can resize image, see https://learn.microsoft.com/zh-cn/dotnet/maui/user-interface/graphics/images#resize-an-image

**Reply by MahmoudAlEssawi** [reply]
I tried Maui.Graphics but didn't work on windows and the image is rotated on android.

I need to compress the image in background without loading it on the screen.

**Reply by xtuzy** [reply]
why not try image library, like skiasharp or imagesharp, or you can search how to do on every platform, such as use java on android, convert java to c# is easy.

**Reply by MahmoudAlEssawi** [reply]
I Tried IImage Class and it works:

        private static IImage ConvertPathToIImage(Stream stream)
        {
            IImage image = null;
            if (stream != null)
            {
#if IOS || ANDROID || MACCATALYST
        // PlatformImage isn't currently supported on Windows.
        image = PlatformImage.FromStream(stream);
#elif WINDOWS
    image = new W2DImageLoadingService().FromStream(stream);
 
#endif
            }

            return image;
        }

        public static IImage ImageDownSize(Stream stream,float MaxWidthOrHeight = 720)
        {
            IImage image = ConvertPathToIImage(stream)??null;

            if (image != null)
            {
                var DownSizedIMG = image.Downsize(MaxWidthOrHeight);
                return DownSizedIMG;
            }
            else
            {
                return null;
            }
             
        }

        public static async Task<ImageObject> SaveIImageToImageObject(IImage image, string imagename)
        {
            if(image == null) return null;
            //var imagename = CloudKeyGeneration.GenerateKey("Image")+".Jpeg";
            var fullpath = Path.Combine(cacheDir, imagename);
            if (File.Exists(fullpath)) File.Delete(fullpath);
            using FileStream localFileStream = File.OpenWrite(fullpath);
#if WINDOWS
await image.SaveAsync(localFileStream, ImageFormat.Jpeg, 0.25f);
#else
            await image.SaveAsync(localFileStream, ImageFormat.Jpeg, 0.8f);
#endif

            ImageObject imageobject = new()
            {
                IMGSource = fullpath,
                LocalFileName = imagename,
                ImageExtention = "Jpeg",
                Length = localFileStream.Length
            };
            return imageobject;
        }
